apps/studio/components/grid/components/header/filter/FilterPopoverNew.tsx

```diff
@@ -0,0 +1,277 @@
+
+export interface FilterPopoverProps {
+  portal?: boolean
+}
+
+// Convert Filter[] to FilterGroup
+// Note: pg-meta Filter allows column to be string | string[] (for tuple filters)
+// We only support simple string columns in the FilterBar
+function filtersToFilterGroup(filters: Filter[]): FilterGroup {
+  return {
+    logicalOperator: 'AND',
+    conditions: filters
+      .filter((filter) => typeof filter.column === 'string') // Skip tuple filters
+      .map((filter) => ({
+        propertyName: filter.column as string,
+        operator: filter.operator,
+        value: filter.value,
+      })),
+  }
+}
+
+// Convert FilterGroup to Filter[]
+function filterGroupToFilters(group: FilterGroup): Filter[] {
+  const filters: Filter[] = []
+  for (const condition of group.conditions) {
+    if (isGroup(condition)) {
+      filters.push(...filterGroupToFilters(condition))
+    } else {
+      filters.push({
+        column: condition.propertyName,
+        operator: condition.operator as Filter['operator'],
+        value: String(condition.value ?? ''),
+      })
+    }
+  }
+  return filters
+}
+
+// Custom date picker component for the FilterBar
+function DatePickerOption({ onChange, onCancel, search }: CustomOptionProps) {
+  const [date, setDate] = useState<Date | undefined>(search ? new Date(search) : undefined)
+
+  return (
+    <div className="w-[300px] space-y-4">
+      <Calendar
+        autoFocus
+        mode="single"
+        defaultMonth={date}
+        selected={date}
+        onSelect={setDate}
+        className="w-full"
+      />
+      <div className="flex justify-end gap-2 py-3 px-4 border-t">
+        <Button type="default" onClick={onCancel}>
+          Cancel
+        </Button>
+        <Button type="primary" onClick={() => onChange(date ? format(date, 'yyyy-MM-dd') : '')}>
+          Apply
+        </Button>
+      </div>
+    </div>
+  )
+}
+
+// Check if column is a date/datetime/time type
+function isDateLikeColumn(column: SupaColumn): boolean {
+  return (
+    isDateColumn(column.format) || isDateTimeColumn(column.format) || isTimeColumn(column.format)
+  )
+}
+
+function serializeFilterProperties(
+  filterProperties: FilterProperty[]
+): SerializableFilterProperty[] {
+  return filterProperties.map((property) => ({
+    label: property.label,
+    name: property.name,
+    type: property.type,
+    operators: property.operators,
+    options: Array.isArray(property.options)
+      ? property.options
+          .map((option) => {
+            if (typeof option === 'string') return option
+            const hasValue = typeof (option as any)?.value === 'string'
+            const hasLabel = typeof (option as any)?.label === 'string'
+            if (hasValue && hasLabel) return (option as any).value || (option as any).label
+            if (hasLabel) return (option as any).label
+            return null
+          })
+          .filter((value): value is string => Boolean(value))
+      : undefined,
+  }))
+}
```

Comment:
All these methods can be moved to a .utl.ts file and be unit tested. We should refactor these methods out of this file and test the values